servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/WriteStreamSubscriber.java

```diff
@@ -248,16 +239,26 @@ public void cancel() {
         }
     }
 
-    private void requestMoreIfRequired(@Nullable Subscription subscription) {
+    private void initialRequestN(Subscription subscription) {
+        if (isClient) {
+            if (channel.isWritable()) {
+                subscription.request(1L);   // Request meta-data only
+            }
+        } else {
+            requestMoreIfRequired(subscription, -1L);
+        }
+    }
+
+    private void requestMoreIfRequired(@Nullable Subscription subscription, long bytesBeforeUnwritable) {
         // subscription could be null if channelWritable is invoked before onSubscribe.
         // If promise is not writable, then we will not be able to write anyways, so do not request more.
         if (subscription == null || subscription == CANCELLED || !promise.isWritable()) {
             return;
         }
 
-        long n = demandEstimator.estimateRequestN(channel.bytesBeforeUnwritable());
+        long n = demandEstimator.estimateRequestN(bytesBeforeUnwritable >= 0 ? bytesBeforeUnwritable :
```

Comment:
Why is the check required? When `bytesBeforeUnwritable` is >= it's equal to `channel.bytesBeforeUnwritable()` (line 164), otherwise, `channel.bytesBeforeUnwritable()` is used directly.

Reply:
The most frequent method that invokes `requestMoreIfRequired` is `doWrite`, where we already have `capacityAfter`. It's a way to avoid the second invocation of `channel.bytesBeforeUnwritable()` that will return the same result because we are on a single event-loop thread here. `bytesBeforeUnwritable()` is not super cheap because it does some math and volatile read internally.